adding mindgraph-cli

sibidass opened the pull request:
mindgraph-cli will read files from command line.

### Sample Usage
`python mindgrap-cli -f filename1.yaml,filename2.yaml`
### ### For help
`python mindgraph-cli -h`

Also added yaml validation using simple load method of pyyaml

Commit:
adding yaml validation using simple pyyaml load

Commit:
PEP8 compliance

capsulecorplab commented on mindgraph-cli.py:
```diff
@@ -0,0 +1,36 @@
+
+def main():
+    args = arg_parser()
+    file_list = args.files.split(",")
```
Could this be a whitespace, instead? I feel like commas aren't as common of a pattern for delimiting multiple command-line arguments

sibidass replied:
@capsulecorplab Problem with whitespace is that, it can cause confusion, if the filename itself has a space. In shell, it will be de-limited by a backslash, but still there will be a space, which python won't be able to differentiate. What do you think?
Still, if you think reading from file is going to be the only functionality of this cli, we can remove -f flag and make it as a positional param.

capsulecorplab commented:
Thanks for the PR, @sibidass!

Aside from the comma delimiter, I also noticed the module is outside the main package folder. Would this affect behavior if installed via pip? I feel like a unit test could accommodate the PR, but up to you.

Changes look good otherwise!

sibidass commented:
@capsulecorplab When importing **mindgraph**, in our case inside mindgraph-cli.py, it will first pull local reference, which is the dir relative to the file path. So even if **mindgraph** module installed through pip, or not installed at all, mindgraph-cli should work.
Sure. Will add a unit test as well, when I get time.

sibidass commented:
@capsulecorplab unlike the packaged module, cli won't be installed, as it is not a module and supposed to run in command line. User need to download the repo and run the cli as it is. Another suggestion is to add a Makefile for installing cli into the system binary path.

yamad commented:
You may want the [nargs](https://docs.python.org/3/library/argparse.html#nargs) argument to `ArgumentParser.add_argument`.

With `parser.add_argument('files', nargs='+',...)`, then `files` takes one or more arguments and saves them as a list.

Commit:
moving mindgraphcli into module path, adding nargs for handling space separated files rather than comma, adding cli support in setup file

Commit:
moving cli inside module path

Commit:
PEP8 standards

sibidass commented:
@yamad nice! Thanks for pointing to **nargs**. I have added nargs into the code. Cli now recognizes space separated files without having to use split and hence no problem in using files having spaces in their names(have to delimit with backslash incase filename itself has space).
@capsulecorplab I have modified the code to accept space separated files. Also moved mindgraph_cli inside the module path, updated setup.py, so that the cli will be installed as part of mindgraph module installation. Updated usage instruction is as given below.

### Usage

- Install the **mindgraph** module using the installation instructions.

- Once module is installed, use the below command to invoke cli.
 **To read yaml files:**
`mindgraphcli -f filename1.yaml filename2.yaml`
**For help:**
`mindgraphcli -h`

Commit:
adding args into arg_parser to speed up operation, adding import for cli in init file

capsulecorplab commented:
Looks like you're getting pep8'd...
```
__________________________________ PEP8-check __________________________________
/home/travis/build/capsulecorplab/mindgraph/mindgraph/mindgraph_cli.py:9:1: E303 too many blank lines (3)
def arg_parser(args):
^
```

Otherwise, changes look great @sibidass!

capsulecorplab committed:
removed extra blank line to satisfy the pep8 gods

sibidass commented:
@capsulecorplab Thanks for satisfying pep8 gods :smiley:
Will add coverage for cli.
